torch/utils/model_dump/code.js

```diff
@@ -0,0 +1,487 @@
+      return "tuple((";
+    }
+    if (data.__module_type__) {
+      return data.__module_type__ + "()";
+    }
+    if (data.__tensor_v2__) {
+      const [storage, offset, size, stride, grad] = data.__tensor_v2__;
+      const [dtype, key, device, numel] = storage;
+      let parts = [
+        "(" + size.join(",") + ")",
+        dtype,
+      ];
+      if (device != "cpu") {
+        parts.push(device);
+      }
+      if (grad) {
+        parts.push("grad");
+      }
+      // TODO: Check stride and indicate if the tensor is channels-last or non-contiguous
+      // TODO: Check size, stride, offset, and numel and indicate if
+      // the tensor doesn't use all data in storage.
+      // TODO: Maybe show key?
+      void(offset);
+      void(stride);
+      void(key);
+      void(numel);
+      return "tensor(" + parts.join(", ") + ")";
+    }
+    throw new Error("TODO: handle dict, etc.");
+  }
+
+  renderBody(indent, data) {
+    if (data === null || this.INLINE_TYPES.has(typeof(data))) {
+      throw "Should not reach here."
+    }
+    if (typeof(data) != "object") {
+      throw new Error("Not an object");
+    }
+    if (Array.isArray(data)) {
+      let new_indent = indent + "\u00A0\u00A0";
+      let parts = [];
+      for (let idx = 0; idx < data.length; idx++) {
+        // Does it make sense to put explicit index numbers here?
+        parts.push(html`<br/><${ModelData} prefix=${idx + ": "} indent=${new_indent} data=${data[idx]} />`);
+      }
+      return parts;
+    }
+    if (data.__tuple_values__) {
+      // Handled the same as lists.
+      return this.renderBody(indent, data.__tuple_values__);
+    }
+    if (data.__module_type__) {
+      const mstate = data.state;
+      if (mstate === null || typeof(mstate) != "object" || !mstate.__is_dict__) {
+        throw new Error("Bad module state");
+      }
+      let new_indent = indent + "\u00A0\u00A0";
+      let parts = [];
+      for (let idx = 0; idx < mstate.keys.length; idx++) {
+        if (typeof(mstate.keys[idx]) != "string") {
+          parts.push(html`<br/>${new_indent}Non-string key`);
+        } else if (this.IGNORED_STATE_KEYS.has(mstate.keys[idx])) {
+          // Do nothing.
+        } else {
+          parts.push(html`<br/><${ModelData} prefix=${mstate.keys[idx] + ": "} indent=${new_indent} data=${mstate.values[idx]} />`);
+        }
+      }
+      return parts;
+    }
+    if (data.__tensor_v2__) {
+      throw "Should not reach here."
+    }
+    throw new Error("TODO: handle dict, etc.");
+  }
+
+  render({data, indent, prefix}, {shown}) {
+    const exp = this.expando(data) ? html`<span class=caret onClick=${() => this.click()} >${caret(shown)} </span>` : "";
+    const headline = this.renderHeadline(data);
+    const body = shown ? this.renderBody(indent, data) : "";
+    return html`${indent}${exp}${prefix}${headline}${body}`;
+  }
+}
+
+function ZipContentsSection({model: {zip_files}}) {
+  // TODO: Add human-readable sizes?
```

Comment:
I think non-human readable is fine since one probably wants to see the actual size of the file.